hydro_cli/src/lib.rs

```diff
@@ -685,39 +685,45 @@ fn null(py: Python<'_>) -> PyResult<Py<PyAny>> {
 
 #[pyclass]
 struct ServerPort {
-    underlying: Option<ServerOrBound>,
+    underlying: hydroflow_cli_integration::ServerPort,
 }
 
 #[pymethods]
 impl ServerPort {
     fn json(&self, py: Python<'_>) -> Py<PyAny> {
-        if let ServerOrBound::Server(server) = self.underlying.as_ref().unwrap() {
-            pythonize(py, &server).unwrap()
-        } else {
-            panic!()
-        }
+        pythonize(py, &self.underlying).unwrap()
     }
 
     #[allow(clippy::wrong_self_convention)]
     fn into_source<'p>(&mut self, py: Python<'p>) -> PyResult<&'p PyAny> {
-        let underlying = self.underlying.take().unwrap();
+        let runtime_read = TOKIO_RUNTIME.read().unwrap();
+        let guard = runtime_read.as_ref().unwrap().enter();
+        let realized = ServerOrBound::Server((&self.underlying).into());
+        drop(guard);
+        drop(runtime_read);
+
         interruptible_future_to_py(py, async move {
             Ok(PythonStream {
                 underlying: Arc::new(RwLock::new(
-                    underlying.connect::<ConnectedBidi>().await.into_source(),
+                    realized.connect::<ConnectedBidi>().await.into_source(),
                 )),
             })
         })
     }
 
     #[allow(clippy::wrong_self_convention)]
     fn into_sink<'p>(&mut self, py: Python<'p>) -> PyResult<&'p PyAny> {
-        let underlying = self.underlying.take().unwrap();
-        interruptible_future_to_py(py, async move {
-            let connected = underlying.connect::<ConnectedBidi>().await.into_sink();
+        let runtime_read = TOKIO_RUNTIME.read().unwrap();
+        let guard = runtime_read.as_ref().unwrap().enter();
+        let realized = ServerOrBound::Server((&self.underlying).into());
+        drop(guard);
+        drop(runtime_read);
```

Comment:
Possible to turn this code into a helper fn? would naturally bound the scope too